clouddriver-ecs/src/test/groovy/com/netflix/spinnaker/clouddriver/ecs/services/ContainerInformationServiceSpec.groovy

```diff
@@ -247,6 +252,199 @@ class ContainerInformationServiceSpec extends Specification {
+      serviceName: serviceName,
+      loadBalancers: [new LoadBalancer()]
+    )
+
+    serviceCacheClient.get(_) >> cachedService
+    taskCacheClient.get(_) >> new Task(lastStatus: "RUNNING", healthStatus: "UNKNOWN")
+    taskDefinitionCacheClient.get(_) >> new TaskDefinition(containerDefinitions:  Lists.newArrayList(new ContainerDefinition(healthCheck: null)))
+    targetHealthCacheClient.get(_) >> new EcsTargetHealth(targetHealthDescriptions: List.of(
+      new TargetHealthDescription(targetHealth: new TargetHealth(state: "unhealthy")),
+      new TargetHealthDescription(targetHealth: new TargetHealth(state: "draining"))
+    ))
+
+    def expectedHealthStatus = [
+      [
+        instanceId: taskId,
+        state     : 'Unknown',
+        type      : type
+      ],
+      [
+        instanceId: taskId,
+        state     : "Down",
+        type      : 'ecs',
+        healthClass: 'platform'
+      ]
+    ]
+
+    when:
+    def retrievedHealthStatus = service.getHealthStatus(taskId, serviceName, 'test-account', 'us-west-1')
+
+    then:
+    retrievedHealthStatus == expectedHealthStatus
+  }
+
+  def 'should return Starting status if task is running but healthcheck in TargetGroups is unknown'() {
+    given:
+    def taskId = 'task-id'
+    def serviceName = 'test-service-name'
+    def type = 'loadBalancer'
+
+    def cachedService = new Service(
+      serviceName: serviceName,
+      loadBalancers: [new LoadBalancer()]
+    )
+
+    serviceCacheClient.get(_) >> cachedService
+    taskCacheClient.get(_) >> new Task(lastStatus: "RUNNING", healthStatus: "UNKNOWN")
+    taskDefinitionCacheClient.get(_) >> new TaskDefinition(containerDefinitions:  Lists.newArrayList(new ContainerDefinition(healthCheck: null)))
+    targetHealthCacheClient.get(_) >> new EcsTargetHealth(targetHealthDescriptions: List.of(
+      new TargetHealthDescription(targetHealth: new TargetHealth(state: "initial")),
+      new TargetHealthDescription(targetHealth: new TargetHealth(state: "draining"))
+    ))
+
+    def expectedHealthStatus = [
+      [
+        instanceId: taskId,
+        state     : 'Unknown',
+        type      : type
+      ],
+      [
+        instanceId: taskId,
+        state     : "Starting",
+        type      : 'ecs',
+        healthClass: 'platform'
+      ]
+    ]
+
+    when:
+    def retrievedHealthStatus = service.getHealthStatus(taskId, serviceName, 'test-account', 'us-west-1')
+
+    then:
+    retrievedHealthStatus == expectedHealthStatus
+  }
+
+  def 'should return Up status if task is running but healthcheck in TargetGroups is healthy'() {
+    given:
+    def taskId = 'task-id'
+    def serviceName = 'test-service-name'
+    def type = 'loadBalancer'
+
+    def cachedService = new Service(
+      serviceName: serviceName,
+      loadBalancers: [new LoadBalancer()]
+    )
+
+    serviceCacheClient.get(_) >> cachedService
+    taskCacheClient.get(_) >> new Task(lastStatus: "RUNNING", healthStatus: "UNKNOWN")
```

Comment:
This doesn't match the test description where heatlh status is "UNKNOWN" ... and if health status is unknown... that seems problematic to mark the server as "HEALTHY"...

Reply:
By default, if the containerInformation service lacks health check details in the containerDefinition, it considers the deployment healthy as long as the lastStatus is set to RUNNING.

In this case, health checks are being reported by the targetHealthcheck, while the containerDefinition does not include any health check information.

that is why the healtStatus is UNKNOWN but it should look for targetHealthchecks if the containerDefinition does not include healthcheck